crypto/crypto/src/hash.rs

```diff
@@ -442,6 +442,10 @@ pub trait CryptoHash {
 
 /// A trait for representing the state of a cryptographic hasher.
 pub trait CryptoHasher: Default + std::io::Write {
```

Comment:
Not sure if w/o actual hashing happening, this should be still called a Hasher or CryptoSerializer or TypedSerializer.

Reply:
The prime goal for `CryptoHasher` is to provide the `update`, `finish` functions of a real honest-to-god hasher, which is invoked when you call `<T as CryptoHash>.hash(self)`.

The fact I'm piggy-backing on it to add non-hashed functionality doesn't mean the structure's hashing purposes get invalidated :)

Reply:
That’s true, but if we never sign hashes and this is only used for type-defined serialization, I don’t see why they should be called hashers vs typedLCS (and then we can use it for other purposes too where we specifically need LCS Serialization with type (domain separation) attached )

Reply:
`CryptoHasher` is usable (and used) independently from LCS.

Reply:
We could create a smaller trait (say `CryptoSeed`) but I'm not sure if it's worth the trouble.